src/neuroglancer/util/geom.ts

```diff
@@ -148,3 +149,35 @@ export function translationRotationScaleZReflectionToMat4(
   out[2] = scale[2] * zReflection;
   return mat4.fromRotationTranslationScale(out, rotation, translation, <vec3>temp);
 }
+
+
+/**
+ * Transforms a z-index to/from Morton Codes (i.e. z-indices).
+ */
+const magicBits = [
+  new Uint64(0x49249249, 0x12492492), // AND step
+  new Uint64(0xc30c30c3, 0x30c30c30), // XOR steps
+  new Uint64(0x0f00f00f, 0xf00f00f0),
+  new Uint64(0xff0000ff, 0x00ff0000),
+  new Uint64(0x0000ffff, 0x00ff0000),
+  new Uint64(0x001fffff, 0x00000000),
+];
+
+export function compactMorton(input: Uint64) {
```

Comment:
In general I think avoiding creating new objects is helpful, though I'm not sure exactly how it affects performance, and this isn't necessarily extremely performance sensitive, so I'm not too worried about it in this case.